Content.Shared/Decals/DecalGridComponent.cs

```diff
@@ -23,7 +24,7 @@ public sealed class DecalGridComponent : Component
         [Serializable, NetSerializable]
         public sealed class DecalChunk
         {
-            [DataField("decals")]
+            [IncludeDataField(customTypeSerializer:typeof(DictionarySerializer<uint, Decal>))]
             public Dictionary<uint, Decal> Decals;
```

Comment:
I don't fully understand why its necessary, but it seems to error if the serializer is not explicitly specified?

Reply:
If I'm reading the code in `SerializationManager` correctly, its intentional that this only works without a custom type serializer if the type is a data definition? Is what I'm doing here cursed for some reason?

Reply:
it should work without the customtypeserializer too. the checks at init just ensure that we dont have a recursion in includedatafields

Reply:
Well without it `SerializationManager` throws a "Invalid Types used for include fields" and afaict its not related to a recursion check